lib/Lower/Mangler.cpp

```diff
@@ -19,66 +19,85 @@
                  llvm::SmallVector<llvm::StringRef, 2> &result) {
-  if (scope->kind() == Fortran::semantics::Scope::Kind::Global) {
+  if (scope.kind() == Fortran::semantics::Scope::Kind::Global) {
     return;
   }
-  moduleNames(&scope->parent(), result);
-  if (scope->kind() == Fortran::semantics::Scope::Kind::Module)
-    if (auto *symbol = scope->symbol())
+  moduleNames(scope.parent(), result);
+  if (scope.kind() == Fortran::semantics::Scope::Kind::Module)
+    if (auto *symbol = scope.symbol())
       result.emplace_back(toStringRef(symbol->name()));
 }
 
 llvm::SmallVector<llvm::StringRef, 2>
-moduleNames(const Fortran::semantics::Scope *scope) {
+moduleNames(const Fortran::semantics::Symbol &symbol) {
+  const auto &scope{symbol.owner()};
   llvm::SmallVector<llvm::StringRef, 2> result;
   moduleNames(scope, result);
   return result;
 }
 
 llvm::Optional<llvm::StringRef>
-hostName(const Fortran::semantics::Scope *scope) {
-  if (scope->kind() == Fortran::semantics::Scope::Kind::Subprogram) {
-    auto &parent = scope->parent();
-    if (parent.kind() == Fortran::semantics::Scope::Kind::Subprogram)
-      if (auto *symbol = parent.symbol()) {
-        return {toStringRef(symbol->name())};
-      }
+hostName(const Fortran::semantics::Symbol &symbol) {
+  const auto &scope{symbol.owner()};
+  if (scope.kind() == Fortran::semantics::Scope::Kind::Subprogram) {
+    assert(scope.symbol() && "subprogram scope must have a symbol");
+    return {toStringRef(scope.symbol()->name())};
   }
   return {};
 }
 
 } // namespace
 
+static const Fortran::semantics::Symbol *
+findInterfaceIfSeperateMP(const Fortran::semantics::Symbol &symbol) {
+  const auto &scope{symbol.owner()};
+  if (symbol.attrs().test(Fortran::semantics::Attr::MODULE) &&
+      scope.IsSubmodule()) {
+    // FIXME symbol from MpSubprogramStmt do not seem to have
+    // Attr::MODULE set.
+    const auto *iface{scope.parent().FindSymbol(symbol.name())};
+    assert(iface && "Separate module proc must be declared");
+    return iface;
+  }
+  return nullptr;
+}
+
 // Mangle the name of `symbol` to make it unique within FIR's symbol table using
 // the FIR name mangler, `mangler`
 std::string
 Fortran::lower::mangle::mangleName(fir::NameUniquer &uniquer,
-                                   const Fortran::semantics::SymbolRef symbol) {
+                                   const Fortran::semantics::Symbol &symbol) {
   return std::visit(
       Fortran::common::visitors{
           [&](const Fortran::semantics::MainProgramDetails &) {
             return uniquer.doProgramEntry().str();
           },
           [&](const Fortran::semantics::SubprogramDetails &) {
-            auto &cb{symbol->name()};
-            auto modNames{moduleNames(symbol->scope())};
-            return uniquer.doProcedure(modNames, hostName(symbol->scope()),
-                                       toStringRef(cb));
+            auto &symbolName{symbol.name()};
```

Comment:
uniform initializers will never generate a warning with `auto`...

Reply:
Thanks, all `auto xx{}` changed to `auto xx = ` in `Mangler.cpp`, I know there are other places in the Bridge where there {} init, I will try to make that consistent when working on related code.

Reply:
We should have some sort of indication (at the top of the file?) that a particular source file isn't using the MLIR coding standard and vice versa. I started adding some comments like that in places where it looked like MLIR was prevailing.